controllers/helmchart_controller.go

```diff
@@ -54,6 +54,7 @@ import (
 	"github.com/fluxcd/pkg/runtime/patch"
 	"github.com/fluxcd/pkg/runtime/predicates"
 	"github.com/fluxcd/pkg/untar"
+	"github.com/hashicorp/go-multierror"
```

Comment:
Let's use `Aggregate` error from k8s.io/apimachinery/pkg/util/errors for multi errors. We already use it in most of the controller code where we need.
An example can be found in https://github.com/fluxcd/source-controller/blob/a00d0edcd75ebc01996dcc94831a0db52367f6ec/controllers/helmrepository_controller_oci.go#L160.